MyBlog.Services/Areas/Users/UserService.cs

```diff
@@ -52,6 +65,11 @@ public async Task<int> UpdateByIdAsync(int id, UserDtoInput userInput)
         var user = await _context.Users.FirstOrDefaultAsync(u => u.Id == id)
                    ?? throw new NotFoundException($"User with Id: {id} is not found");
 
+        if (CheckForFreeAsync(id, userInput).Result == false)
+        {
```

Comment:
it`s better to call methods and save it in variables, and that
call if statement with var